### DISCUSSION TITLE
F-droid repo

### DISCUSSION THREAD

**Discussion by Adithyan007hack** [libre-tube/LibreTube#463] [General]
### Describe your suggested feature

Why don't libretube have an f-droid repo?
The updates arrive much slow in the default f-droid universal repo
as the new pipe has its own repo for faster update 
If there was an f-droid repo the updates could arrive faster right??

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [x] I have written a short but informative title.
- [x] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
we would need a server to host it and that would cost money

**Comment by Adithyan007hack** [comment]
> 

"It is okay I just asked due to curiosity"
I can wait for a few days to receive updates please don't abandon this project

**Comment by Bnyro** [comment]
It's fine, it is a good idea but we would need donations for the server hosting for it, I already thought about it too :D

**Comment by Bnyro** [comment]
I still want it to be there :D

**Comment by Bnyro** [comment]
We just have to find someone who can host it

**Comment by ghost** [comment]
I wonder how to find it?😁

**Comment by Bnyro** [comment]
@Nizami20052022 idk xD, maybe with donations ( if there were some)

**Comment by Bnyro** [comment]
Actually we can't use github.io, that's how it works https://f-droid.org/docs/Setup_an_F-Droid_App_Repo/
we would need our own server for it

**Comment by Bnyro** [comment]
or we could ask @ IzzySoft whether he could accept us for his repo, his repo updates itself really fast

**Comment by Bnyro** [comment]
But it looks like he only accepts apps that are not on F-Droid yet

**Comment by Adithyan007hack** [comment]
> 

Don't worry bro we will find someone for the server hosting
and I have completed the Malayalam translation as much as I can, and it is now 90% complete

**Comment by Bnyro** [comment]
@Adithyan007hack That's great :)

**Comment by bitplane** [comment]
I haven't checked, but does it need to be always running? It looks like it supports mirrors so should be possible using github actions to generate the repo and commit it to a jeckyll site that holds the repo. Only real question is whether the URLs are expected to be inside each other or it can be a fake directory list with links to releases for each file.

**Comment by shuvashish76** [comment]
> But it looks like he only accepts apps that are not on F-Droid yet

No such [inclusion criteria](https://gitlab.com/IzzyOnDroid/repo/#what-are-the-requirements-an-app-must-meet-to-be-included-with-the-repo) for izzyrepo.

> we could ask @ IzzySoft whether he could accept us for his repo, his repo updates itself really fast

Ping @IzzySoft for help, regarding this app inclusion & custom F-Droid repo related info...

**Comment by IzzySoft** [comment]
LibreTube is already available at F-Droid, so what's the reason you want it in my repo as well? Of course it is technically possible (and there are several other apps listed in both), but I'd like to know the reason. Checking its presence:

```
$ iod shell fda com.github.libretube
LibreTube (com.github.libretube) has 3 packages at main, last one added at 2022-06-09.
LibreTube (com.github.libretube) has 2 packages at archive, last one added at 2022-03-21.
Package Id 'com.github.libretube' not found in 'iod'.
```

it's even updated recently (just 2 days ago).

I have a vague hunch it might be about "really fast updates" whenever you had to work around something YT just broke. But unless you're using reproducible builds at F-Droid (as e.g. NewPipe does), there's no cross-updating possible; users then had to make their choice of source from the start.

That said, if you want to I can of course add LibreTube back to my repo (where it was present until this year's April until it became available at F-Droid itself), metadata are still archived and can be easily restored.

@Bnyro for easy setup of your own repo, if you still want to pursue that path, you can also see my articles on the topic – starting with [Setting up a simple binary F-Droid repo](https://android.izzysoft.de/articles/named/fdroid-simple-binary-repo) (oops, I just see I forgot to translate the [follow-up article](https://android.izzysoft.de/articles/named/fdroid-simple-binary-repo-2) to English).

**Comment by Bnyro** [comment]
@IzzySoft First of all thank you for your answer, I really appreciate your help.
We currently don't depend on the "really fast updates" because we use the Piped proxy, so we depend on them to fix things YouTube breaks. However, we would really appreciate to become part of your repo again if you don't mind since it takes F-Droid usually more than a week to update the app. 
Lastly, I will go through your articles about settings up a repo. It's perfect that they are available in german, it'll make my life a lot easier :D The thing is that we still neither have our own server nor a LibreTube domain so that path will probably take some time.
Greetings and thanks for your help

**Comment by IzzySoft** [comment]
In this case: welcome back to IzzyOnDroid! LibreTube will be available there again with the next sync in less than 4h. As before, I've picked the `armeabi-v7a` as that's the most common architecture (and even understood by v8 and arm64 AFAIK, at least with currently existing devices).

May I suggest a change to your `full_description.txt` by the way? Here's what I've set up with IoD:

```html
<i>LibreTube</i> is an alternative frontend for YouTube. It uses the <a href='https://github.com/TeamPiped/Piped' target='_blank' rel='nofollow'>Piped</a> API to load data and play videos and thus needs no YT account and no Google Services on your device.
```

Basically says the same, but looks nicer and directly links to Piped for reference (and yes, this works with F-Droid as well – only Play might strip the links, but that's of no consequence for a "forbidden app", right? :rofl:)

PS: Per-release changelogs are missing in your fastlane since `versionCode:5`, you might wish to add them again starting with the next release (should you add the current one as well, drop me a note and I pull it manually).

**Comment by Bnyro** [comment]
@IzzySoft Thank you for bringing LibreTube back to your repo :D
I applied your suggestions to the full_description and added the missing changelog for the last version, I will start adding them from now on. Thank you for the hints and your help :)

**Comment by IzzySoft** [comment]
Thanks @Bnyro! I've just pulled the changelog (a bit too late for today's sync, but will become visible tomorrow) and also adjusted metadata here to pull fulldesc from your fastlane as well from now on.

**Comment by IzzySoft** [comment]
PS: No need to hurry with your own F-Droid repo if you're satisfied with the current situation :wink:

**Comment by Bnyro** [comment]
@IzzySoft Wow, your repo also includes the new logo now, that's great. You're really doing a great job helping open source apps to grow and take them off some work, great thanks <3

**Comment by IzzySoft** [comment]
Thanks for your kind words, @Bnyro – that motivates me to keep up doing so :hugs:

**Comment by Adithyan007hack** [comment]
Hey guys, I have a question 
I have already installed the fdroid version of the app version 0.3.0, do in need to uninstall it before installing the new version 0.3.1 from the izzyondroid repo.

**Reply by IzzySoft** [reply]
Most likely,due to different signing keys being used.

**Comment by Bnyro** [comment]
@Adithyan007hack yes, will you have to reinstall it once

**Reply by Adithyan007hack** [reply]
> @Adithyan007hack yes, will you have to reinstall it once

Is there a way to export subscriptions from libretube

**Reply by Bnyro** [reply]
not yet, you have to visit the webpage of the instance you use and export them there

**Reply by Adithyan007hack** [reply]
> you have to visit the webpage of the instance you use and export them there

Didn't understand, can you just clarify what you are saying

**Reply by Bnyro** [reply]
If you use kavin.rocks as instance for example then go to piped.kavin.rocks, login, go to feed, subscriptions, then export

**Comment by Adithyan007hack** [comment]
In gow much time the new version will arrive at izzyondroid repo
Because i have been wating for the version 0.3.2 to arrive but it is bot yet there it has been listed on izzy's website but not yet arrived in fdroid and droidfy

**Reply by shuvashish76** [reply]
> it has been listed on izzy's website but not yet arrived in fdroid and droidfy

It's live in the droidify app. BTW if any app updated on izzy website but not on your app then it's your F-Droid client issue.

**Reply by IzzySoft** [reply]
The app always looks at its local index, which is only updated in intervals (by default: 24h). In the official app you can check tapping the F-Droid entry in *Options › Repositories* and look at the date. If it's outdated, go back to either the "Tasks" or the "Newest" view and "pull-to-refresh".

**Reply by Adithyan007hack** [reply]
Refreshed both in fdroid and droidfy but the new version is not listing there and I dont know why

**Reply by Bnyro** [reply]
You have to add the IzzyOnDroid repo to your package sources (can be found under the 3-dot menu in Droidify)

**Reply by Adithyan007hack** [reply]
Already done that 😔 still new version is not listing

**Reply by Adithyan007hack** [reply]
Bro Its just libretube not getting the latest update from izzyondroid all other apps are getting the new update

**Reply by Bnyro** [reply]
It's updated for me on Droidify, you're probably doing sth wrong

**Reply by Adithyan007hack** [reply]
![Screenshot_2022-06-19-22-21-48-596_com.looker.droidify.jpg](https://user-images.githubusercontent.com/87800041/174491860-e7bb36c5-215e-406a-8e24-619a5027f357.jpg)

![Screenshot_2022-06-19-22-22-22-142_com.looker.droidify.jpg](https://user-images.githubusercontent.com/87800041/174491874-a62bd4a2-1cd1-4900-9135-8f5417d96f2c.jpg)

**Reply by shuvashish76** [reply]
No need to show the proof, we believe you. Maybe ask in their [Telegram group](https://t.me/neo_android_store) as this is not related to LibreTube.

**Reply by Adithyan007hack** [reply]
But its not updating even in fdroid so i think it is not a droidfy specific issue and   
I don't know what to do 😭

**Comment by Adithyan007hack** [comment]
**Pls don't get offended** I am just saying my situation
I am the one who suggested for a repo for faster updates and after accomplishing a repo I am unable to update due to an incompatible version
![gg](https://user-images.githubusercontent.com/87800041/174621771-23f20a01-3e5d-4714-b0a5-340eeccf1a03.jpg)

@IzzySoft can you pls explain to me why this happened

**Reply by IzzySoft** [reply]
Oops? Checking… Oh, typo in my config and it pulled the wrong APK (most likely the arm64 as it's the first one in the list – yupp, that one). Fixed that, so the next version should be pulled correctly. Replaced the current version and triggered a manual sync, should be up in 5 minutes. Please refresh the index once more then and you should be able to get the update. (Sync is through now, so just go ahead in the client. Website will show the change in about 10 minutes, but the client already gets the latest changes)

Thanks for checking and spotting the culprit! A single letter was missing and that made all the difference…

**Reply by Adithyan007hack** [reply]
Thank you very much bro😊

**Reply by IzzySoft** [reply]
Thanks to you for digging that up!

**Comment by mychan14** [comment]
Please optimize the horizontal screen display function of the tablet

**Comment by trymeouteh** [comment]
I would like this since when disabling the Piped proxy, I assume this app uses the NewPipe extractor which will stop working every now and then and will not continue to work again until an update has been made. Newpipe app has its own F-Droid repo for quick updates.

**Reply by FireMasterK** [reply]
No it doesn't, you just fetch images and videos directly from Google. Fetching from YouTube directly via NewPipeExtractor will likely never be supported.

**Comment by moteb757** [comment]
ل
